python/pyiceberg/catalog/__init__.py

```diff
@@ -162,6 +163,9 @@ def __init__(self, name: str, **properties: str):
         self.name = name
         self.properties = properties
 
+    def _load_file_io(self, properties: Properties = EMPTY_DICT) -> FileIO:
+        return load_file_io({**self.properties, **properties})
```

Comment:
I don't think that we include table properties in the JVM version, but this seems like a good idea to me!